dev-libs/libcdio: ModernC fixes

immolo opened the pull request:
Lack of contact from upstream so adding Parona's patch locally as tested as working fine.

Closes: https://bugs.gentoo.org/918988
Thanks-to: Alfred Wingate <parona@protonmail.com>


---

Please check all the boxes that apply:

- [X] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [X] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [X] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [X] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

fordfrog commented:
@immolo thanks! i made it straight to stable and dropped the old one. i also updated eapi to 8.

immolo commented:
> @immolo thanks! i made it straight to stable and dropped the old one. i also updated eapi to 8.

No worries,  I've made contact with upstream as well now so I'll drop you an update once we have squared this fix as a permanent fix.
